mathics/builtin/assignment.py

```diff
@@ -792,10 +800,130 @@ def rhs(expr):
 
     def format_definition_input(self, symbol, evaluation):
         'InputForm: Definition[symbol_]'
-
         return self.format_definition(symbol, evaluation, grid=False)
 
 
+
+
+
+
+
+def information_interpret_doc_string(ds):
+    # TODO: work a little bit the interpretation
+    ds = ds.split("\n")
+    out = ""
+    for s in ds:
+        if s[0:6]=="    >>":
+            # If we reach the test section, stop the loop.
+            break
+        out = out + "\n" + s
+    return out
+
+
+def _get_usage_string(symbol,evaluation):
+    '''
+    Returns a python string with the documentation associated to a given symbol.
+    '''
+    definition = evaluation.definitions.get_definition(symbol.name)
+    ruleusage = definition.get_values_list('messages')
+    usagetext = None;
+    from mathics.builtin import builtins
+    bio = builtins.get(definition.name)
+    if bio is not None:
+        usagetext = information_interpret_doc_string(bio.__class__.__doc__)
+
+    # For built-in symbols, looks for a docstring.
+#    if symbol.function. is Builtin:            
+    #evaluation.print_out(String("found: " + usagetext))
+    #usagetext = information_interpret_doc_string(symbol.__doc__)
+        
+    # Looks for the "usage" message. For built-in symbols, if there is an "usage" chain, overwrite the __doc__ information.
+    for rulemsg in ruleusage:
+        if  rulemsg.pattern.expr.leaves[1].__str__()=="\"usage\"":
+            usagetext = rulemsg.replace.value        
+    return  usagetext            
+
+
+
+class Information(PrefixOperator):    
+    """
+    <dl>
+    <dt>'Information[$symbol$]'
+        <dd>Prints information about a $symbol$
+    </dl>
+    'Information' does not print information for 'ReadProtected' symbols.
+    'Information' uses 'InputForm' to format values.
+
+
+
+    >> ?? Plus
+     |
+     |
+     |    <dl>
+     |    <dt>'Plus[$a$, $b$, ...]'</dt>
+     |    <dt>$a$ + $b$ + ...</dt>
+     |    <dd>represents the sum of the terms $a$, $b$, ...
+     |    </dl>
+     = Null
+
+    >> Information[Integrate]
+     |
+     |
+     |    <dl>
+     |    <dt>'Integrate[$f$, $x$]'
+     |    <dd>integrates $f$ with respect to $x$. The result does not contain the additive integration constant.
+     |    <dt>'Integrate[$f$, {$x$, $a$, $b$}]'
+     |    <dd>computes the definite integral of $f$ with respect to $x$ from $a$ to $b$.
+     |    </dl>
+     | 
+     |    Integrate a polynomial:
+     = Null
+
+    >> a = 2;
+    >> Information[a]
+     = Null
+   
+    >> F[x_] := x^2
+    >> F::usage = "F[x] evaluates the square of x."
+    >> H[F[y_,u_]]^:=F[H[y],u]
+    >> Information[F]
+     |F[x] evaluates the square of x
+     = Null
+    """
+    operator="??"
+    precedence=5001
```

Comment:
I do not know if there is an standard value for the precedence of these operators. I just choose a very large one, because I supposed that in the context where these operators make sense, they should be evaluated with very high priority.